### PR TITLE
Phi3 MoE cuda kernel

### PR DESCRIPTION
**Pull request by wangyems**

### CODE REVIEW

File: docs/ContribOperators.md [hunk 1]
```diff
@@ -4394,7 +4397,7 @@ This version of the operator has been available since version 1 of the 'com.micr
 </dl>
 
 
-### <a name="com.microsoft.QMoE"></a><a name="com.microsoft.qmoe">**com.microsoft.QMoE**</a>
+### <a name="com.microsoft.QMoE4Bits"></a><a name="com.microsoft.qmoe4bits">**com.microsoft.QMoE4Bits**</a>
```
Comment (tianleiwu) [inline docs/ContribOperators.md:4400]: This is not backward compatible.
Shall we just add an attribute to QMoE to indicate whether it is 4 bit, 8 bit int quantization?

File: docs/ContribOperators.md [hunk 2]
```diff
@@ -4457,6 +4460,71 @@ This version of the operator has been available since version 1 of the 'com.micr
...
+#### Attributes
+
+<dl>
+<dt><tt>activation_type</tt> : string</dt>
+<dd>Activation function to use. Choose from relu, gelu, silu and identity. Default is relu</dd>
+<dt><tt>k</tt> : int</dt>
+<dd>Number of top experts to select from expert pool</dd>
+<dt><tt>normalize_routing_weights</tt> : int</dt>
+<dd>Whether to normalize routing weights</dd>
+<dt><tt>use_sparse_mixer</tt> : int</dt>
```
Comment (tianleiwu) [inline docs/ContribOperators.md:4480]: why 4 bit does not have this attribute?

File: docs/ContribOperators.md (hunk 1 above)
Comment (wangyems) [reply docs/ContribOperators.md:4400]: let me change back to QMoE with attribute. the initial expectation was to add these two ops in before the release

File: onnxruntime/test/python/transformers/test_parity_phi3_moe.py [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+def print_tensor(name, numpy_array):
+    print(f"const std::vector<float> {name} = {value_string_of(numpy_array)};")
+
+
+def quant_dequant(weights, quant_mode: bool = True):
+    # use the test version `_symmetric_...` to get the non-interleaved weights
+    type = torch.quint4x2 if quant_mode else torch.int8
+    # This import is needed to use torch.ops.trtllm._symmetric_quantize_last_axis_of_batched_matrix()
+    # Comment out this line for passing the lintrunner check in the CI.
+    # import tensorrt_llm
```
Comment (tianleiwu) [inline onnxruntime/test/python/transformers/test_parity_phi3_moe.py:46]: If `import tensorrt_llm` is needed, this test shall fail in CI pipeline?

Comment (wangyems) [reply onnxruntime/test/python/transformers/test_parity_phi3_moe.py:46]: the test itself shall not fail. I decided not to add the trt_llm dependency to CI machine as of now due to dependency conflict.

**Comment by tianleiwu** [comment]
Test failed in A10: ```ort_fastertransformer::generic_moe_gemm_kernelLauncher occupancy > 0 was false. GPU lacks the shared memory resources to run GroupedGEMM kernel```

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**